drivers/spi/spi_ambiq_spis.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (c) 2024 Ambiq <www.ambiq.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ambiq_spis
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_ambiq_spis);
+
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/pm/device_runtime.h>
+
+#include <stdlib.h>
+#include <errno.h>
+#include "spi_context.h"
+#include <am_mcu_apollo.h>
+
+#define PWRCTRL_MAX_WAIT_US 5
+
+typedef int (*ambiq_spi_pwr_func_t)(void);
+
+struct spi_ambiq_config {
+	const struct gpio_dt_spec int_gpios;
+	uint32_t base;
+	int size;
+	const struct pinctrl_dev_config *pcfg;
+	ambiq_spi_pwr_func_t pwr_func;
+	void (*irq_config_func)(void);
+};
+
+struct spi_ambiq_data {
+	struct spi_context ctx;
+	am_hal_ios_config_t ios_cfg;
+	void *ios_handler;
+	int inst_idx;
+	struct k_sem spim_wrcmp_sem;
+};
+
+#define AMBIQ_SPIS_TX_BUFSIZE_MAX 1023
+uint8_t g_pui8TxFifoBuffer[AMBIQ_SPIS_TX_BUFSIZE_MAX];
+
+#define SPI_WORD_SIZE 8
```

Comment:
Add AMBIQ_SPIS_ prefix to all your local defines, as you did on some already, for better readability and consistency